lighthouse-core/audits/byte-efficiency/offscreen-images.js

```diff
@@ -0,0 +1,136 @@
+
+const IGNORE_THRESHOLD_IN_BYTES = 2048;
+
+class OffscreenImages extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Images',
+      name: 'offscreen-images',
+      description: 'Offscreen images',
+      helpText: 'Images that are not above the fold should be lazy loaded on interaction. ' +
+        'Consider using the [IntersectionObserver](https://developers.google.com/web/updates/2016/04/intersectionobserver) API.',
+      requiredArtifacts: ['ImageUsage', 'ViewportDimensions', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {!ClientRect} imageRect
+   * @param {{innerWidth: number, innerHeight: number}} viewportDimensions
+   * @return {number}
+   */
+  static computeVisiblePixels(imageRect, viewportDimensions) {
+    const innerWidth = viewportDimensions.innerWidth;
+    const innerHeight = viewportDimensions.innerHeight;
+
+    const top = Math.max(imageRect.top, -1 * ALLOWABLE_OFFSCREEN_Y);
+    const right = Math.min(imageRect.right, innerWidth + ALLOWABLE_OFFSCREEN_X);
+    const bottom = Math.min(imageRect.bottom, innerHeight + ALLOWABLE_OFFSCREEN_Y);
+    const left = Math.max(imageRect.left, -1 * ALLOWABLE_OFFSCREEN_X);
+
+    return Math.max(right - left, 0) * Math.max(bottom - top, 0);
+  }
+
+  /**
+   * @param {!Object} image
+   * @param {{innerWidth: number, innerHeight: number}} viewportDimensions
+   * @return {?Object}
+   */
+  static computeWaste(image, viewportDimensions) {
+    const url = URL.getDisplayName(image.src, {preserveQuery: true});
+    const totalPixels = image.clientWidth * image.clientHeight;
+    const visiblePixels = this.computeVisiblePixels(image.clientRect, viewportDimensions);
+    const wastedRatio = 1 - visiblePixels / totalPixels;
+    const totalBytes = image.networkRecord.resourceSize;
+    const wastedBytes = Math.round(totalBytes * wastedRatio);
+
+    if (!Number.isFinite(wastedRatio)) {
+      return new Error(`Invalid image sizing information ${url}`);
+    }
+
+    return {
+      url,
+      preview: {
+        url: image.networkRecord.url,
+        mimeType: image.networkRecord.mimeType
+      },
+      totalBytes,
+      wastedBytes,
+      wastedPercent: 100 * wastedRatio,
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {{results: !Array<Object>, tableHeadings: Object,
+   *     passes: boolean=, debugString: string=}}
+   */
+  static audit_(artifacts) {
+    const images = artifacts.ImageUsage;
+    const viewportDimensions = artifacts.ViewportDimensions;
+
+    let debugString;
+    const resultsMap = images.reduce((results, image) => {
+      if (!image.networkRecord) {
+        return results;
+      }
+
+      const processed = OffscreenImages.computeWaste(image, viewportDimensions);
+      if (processed instanceof Error) {
+        debugString = processed.message;
+        return results;
+      }
+
+      // Don't warn about an image that was also used appropriately
```

Comment:
what about something like, "if an image was used more than once, warn about the least wasteful usage of it"